rotkehlchen/utils/snapshots.py

```diff
@@ -0,0 +1,98 @@
+from csv import DictReader
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants.assets import A_USD
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.errors.price import NoPriceForGivenTimestamp
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.price import PriceHistorian
+from rotkehlchen.serialization.deserialize import deserialize_timestamp
+from rotkehlchen.types import Price, Timestamp
+from rotkehlchen.user_messages import MessagesAggregator
+
+
+def validate_import_data(
+        balances_data: List[Dict[str, str]],
+        location_data: List[Dict[str, str]],
+) -> Tuple[bool, str]:
+    """Validates the snapshot data about to be imported."""
+    # check if the headers match the type stored in the db
+    has_invalid_headers = (
+        tuple(balances_data[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
+        tuple(location_data[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
+    )
+    if has_invalid_headers:
+        return False, 'csv file has invalid headers'
+
+    # check if the timestamp can be converted to int
+    try:
+        balances_timestamps = [deserialize_timestamp(entry['timestamp']) for entry in balances_data]  # noqa: E501
+        location_data_timestamps = [deserialize_timestamp(entry['timestamp']) for entry in location_data]  # noqa: E501
+    except DeserializationError:
+        return False, 'csv file contains invalid timestamp format'
+    # check if all timestamps are the same.
+    has_different_timestamps = (
+        balances_timestamps.count(balances_timestamps[0]) != len(balances_timestamps) or
+        location_data_timestamps.count(location_data_timestamps[0]) != len(location_data_timestamps) or  # noqa: E501
+        balances_timestamps[0] != location_data_timestamps[0]
+    )
+    if has_different_timestamps is True:
+        return False, 'csv file has different timestamps'
+
+    return True, ''
+
+
+def parse_import_snapshot_data(
+        balances_snapshot_file: Path,
+        location_data_snapshot_file: Path,
+) -> Tuple[bool, Dict[str, Any]]:
+    """This function takes a path to the snapshot and converts it to a list of dictionaries.
+    It runs a series of validation checks, returns the status of the validation
+    and a dictionary containing an error message and the converted snapshot data.
+    """
+    balances_list = _csv_to_dict(balances_snapshot_file)
+    location_data_list = _csv_to_dict(location_data_snapshot_file)
+    is_valid, message = validate_import_data(
+        balances_data=balances_list,
+        location_data=location_data_list,
+    )
+    if is_valid is False:
+        return False, {'msg': message, 'data': {}}
+
+    return True, {
```

Comment:
Why do you need to return a fixed size dictionary here?

Why not use the pattern we use elsewhere in the code with tuples? The typing will also be better and will enable mypy to check the typing.

So return Tuple[str, List[Balances] List[LocatoinData], ]

Succes or failure can be determined from existence or not of one of the above. Probably the error message.